### ISSUE TITLE
TestUtils fails to run unless TestRunner is also called in 1.4.1

### ISSUE THREAD

**Issue by prometheanfire** [ansible/ansible#5224]
We (gentoo) exclude tests that call the network, because of this we exclude TestRunner and TestPlayBook.py tests from running.  In trying to update ansible in our package tree to 1.4.1 we noticed tests were failing.  Without the code below (from TestRunner.py) being run we get a failure (Unicode error).

```
======================================================================
ERROR: test_check_conditional_jinja2_unicode (TestUtils.TestUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/app-admin/ansible-1.4.1/work/ansible-1.4.1/test/TestUtils.py", line 190, in test_check_conditional_jinja2_unicode
    assert(ansible.utils.check_conditional(u'jinja2_compare "\u00df"', '/', {}) == True)
  File "/var/tmp/portage/app-admin/ansible-1.4.1/work/ansible-1.4.1/lib/ansible/utils/__init__.py", line 183, in check_conditional
original = str(conditional).replace("jinja2_compare ","")
UnicodeEncodeError: 'ascii' codec can't encode character u'\xdf' in position 1: ordinal not in range(128)

----------------------------------------------------------------------
```

In TestRunner.py (around line 106)

```
    def test_copy(self):
        input_ = self._get_test_file('sample.j2')
        output = self._get_stage_file('sample.out')
        result = self._run('copy', [
            "src=%s" % input_,
            "dest=%s" % output,
        ])
```

**Comment by jctanner** [comment]
_PASS_
nosetests --pdb -v test/TestRunner.py test/TestUtils.py

_FAIL_
nosetests --pdb -v test/TestUtils.py

In the situation which the test passes, the environment's default encoding is set to utf-8:

encoding:utf8
type:<type 'unicode'>

In the failing test, the evironment is ascii:

encoding:ascii
type:<type 'unicode'>

I figured this out by adding some debug logging to ansible/utils/**init**.py:

```
 184         #DEBUG
 185         open("/tmp/awx.log", "a").write("encoding:%s\n" % sys.getdefaultencoding())
 186         open("/tmp/awx.log", "a").write("type:%s\n" % type(conditional))
```

So something in ansible is setting the default encoding:

jtanner@u1304:~/ansible$ find . -type f | xargs fgrep -iH 'setdefaultencoding'
Binary file ./lib/ansible/runner/action_plugins/copy.pyc matches
./lib/ansible/runner/action_plugins/copy.py:sys.setdefaultencoding("utf8")
./lib/ansible/runner/action_plugins/unarchive.py:sys.setdefaultencoding("utf8")

That should explain why running the test_copy in TestRunner resolves this problem. To verify this is the issue, I altered the unicode test slightly:

```
    def test_check_conditional_jinja2_unicode(self):

        import sys
        reload(sys)
        sys.setdefaultencoding("utf8")

        assert(ansible.utils.check_conditional(
            u'jinja2_compare "\u00df"', '/', {}) == True)
        assert(ansible.utils.check_conditional(
            u'jinja2_compare var == "\u00df"', '/', {'var': u'\u00df'}) == True)
```

Once I make that change, the test passes in the way you are trying to run it.

```
jtanner@u1304:~/ansible$ nosetests --pdb -v test/TestUtils.py 
test_check_conditional_jinja2_expression (TestUtils.TestUtils) ... ok
test_check_conditional_jinja2_expression_in_variable (TestUtils.TestUtils) ... ok
test_check_conditional_jinja2_literals (TestUtils.TestUtils) ... ok
test_check_conditional_jinja2_unicode (TestUtils.TestUtils) ... ok
test_check_conditional_jinja2_variable_literals (TestUtils.TestUtils) ... ok
test_loaders_expanduser_each_dir (TestUtils.TestUtils) ... ok
test_parse_kv_basic (TestUtils.TestUtils) ... ok
test_template_basic (TestUtils.TestUtils) ... ok
test_template_ds_basic (TestUtils.TestUtils) ... ok
test_template_unicode (TestUtils.TestUtils) ... ok
test_template_whitespace (TestUtils.TestUtils) ... ok
test_varReplace_var_complex_var (TestUtils.TestUtils) ... ok

----------------------------------------------------------------------
Ran 12 tests in 0.044s

OK
```

The test has not been changed since it was created back in July, so I have to wonder if your testing method worked at any point in hisory.

```
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 189)     def test_check_conditional_jinja2_unicode(self):
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 190)         assert(ansible.utils.check_conditional(
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 191)             u'jinja2_compare "\u00df"', '/', {}) == T
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 192)         assert(ansible.utils.check_conditional(
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 193)             u'jinja2_compare var == "\u00df"', '/', {
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 194) 
a1503c4d (Stoned Elipot          2013-07-23 22:24:53 +0200 195) 
```

These are my test results:

1.4.1: fails
1.4.0: fails
1.3.4: passes
1.3.3: passes

**Comment by jctanner** [comment]
closed per #5225
